File: backend/src/main/java/com/sknwl/shareknowledge/domain/entity/Content.java
package com.sknwl.shareknowledge.domain.entity;

import com.sknwl.shareknowledge.domain.entity.enums.ContentType;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
public class Content {
    private Long id;
    private String title;
    private String description;
    private ContentType contentType;
    private String url;
    private Source source;
    private List<String> authors;
    private List<String> subjects;
    private StudyField studyField;
    private Language language;
    private Long durationMinutes;
    private Member publisher;
    private LocalDateTime publishedDateTime;
    private Long reviewers;
    private Double rating;
    private ContentPrice price;
    private List<Comment> comments;
    private CoverImage coverImage;
}

File: frontend/sknwl/src/app/content/content.ts
import { Member } from "../core/member";
import { ContentType } from "./content-type";
import { Cover } from "./cover";

export interface Content {
    id: number,
    title: string,
    description: string,
    contentType: ContentType
    url: string
    source: {
        id: number,
        name: string,
        webSiteUri: string,
        contentTypes: ContentType[]
    },
    authors: string[],
    subjects: string[],
    studyField: {
        id?: number,
        name: string
    },
    language: {
        id: number,
        code: string,
        name: string
    },
    durationMinutes: number,
    publisher: Member,
    publishedDateTime: Date,
    reviewers: number,
    rating: number,
    price: number,
    coverImage: Cover
}
